pkg/operator/ceph/cluster/controller.go

```diff
@@ -450,44 +435,24 @@ func (c *ClusterController) onUpdate(oldObj, newObj interface{}) {
 
 	// attempt to update the cluster.  note this is done outside of wait.Poll because that function
 	// will wait for the retry interval before trying for the first time.
-	done, _ := c.handleUpdate(newClust.Name, cluster)
+	done, _ := cluster.createInstance(c.rookImage, cluster.Info.CephVersion, cephv1.ClusterStateCreating, newClust.Name)
 	if done {
```

Comment:
Instead of returning a new `done` bool from `createInstance()`, could we just return if `err != nil`? Then the signature of `createInstance()` doesn't need to change